### PR TITLE
Additional whitespacing rules

### PR DESCRIPTION
**Pull request by mattBrzezinski**
Addresses #53

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -611,6 +611,85 @@ xy = foo(x, y=3)
...
+    # No
+    if foo
+        println("Hi")
+    end
+    for i in 1:10
+        println(i)
+    end
+    ```
+
+- After a function definition, and before an end statement do not include a line break
```
Comment (mjram0s) [inline README.md:651]:
```suggestion
- After a function definition, and before an end statement do not include a line break.
```

File: README.md [hunk 2]
```diff
@@ -611,6 +611,85 @@ xy = foo(x, y=3)
...
+    end
+
+    # No
+    function foo(bar::In64, baz::Int64)
+        return bar + baz
+
+    end
+    ```
+
+- Use line breaks between control flow statements and returns
```
Comment (mjram0s) [inline README.md:672]:
```suggestion
- Use line breaks between control flow statements and returns.
```

Comment (nicoleepp) [reply README.md:672]: This is supposed to be optional? Can we specifiy that?

Comment (mattBrzezinski) [reply README.md:672]: From #54 this is not optional. It's only optional for single line statements preceding it, e.g.

```julia
# Optional
function foo(bar)
    bar = bar + 1
    baz = bar * 3

    return (bar, baz)
end

# Acceptable
function foo(bar)
    bar = bar + 1
    baz = bar * 3
    return (bar, baz)
end
```

File: README.md [hunk 3]
```diff
@@ -611,6 +611,85 @@ xy = foo(x, y=3)
...
+
+    baz = 3
+    ```
+
+- Use blank-lines to separate different multi-line blocks.
+
+    ```julia
+    # Yes
+    if foo
+      println("Hi")
```
Comment (omus) [inline README.md:635]: I don't believe the 2-space indentation was done on purpose
```suggestion
        println("Hi")
```

File: README.md (hunk 1 above)
Comment (omus) [reply README.md:651]:
```suggestion
- After a function definition, and before an end statement do not include a blank line.
```

File: README.md [hunk 4]
```diff
@@ -611,6 +611,85 @@ xy = foo(x, y=3)
...
+    # Yes
+    function foo(bar; verbose=false)
+        if verbose
+            println("baz")
+        end
+
+        return bar
+    end
+
+    # No
```
Comment (omus) [inline README.md:684]:
```suggestion
    # Ok:
```

File: README.md [hunk 5]
```diff
@@ -611,6 +611,85 @@ xy = foo(x, y=3)
...
+    function foo(bar::In64, baz::Int64)
+        return bar + baz
+
+    end
+    ```
+
+- Use line breaks between control flow statements and returns
+
+    ```julia
+    # Yes
```
Comment (omus) [inline README.md:675]:
```suggestion
    # Yes:
```

File: README.md [hunk 6]
```diff
@@ -611,6 +611,85 @@ xy = foo(x, y=3)
...
 
+- Group similar one line statements together.
+
+    ```julia
+    # Yes
+    foo = 1
+    bar = 2
+    baz = 3
+
+    # No
```
Comment (omus) [inline README.md:622]:
```suggestion
    # No:
```

File: README.md [hunk 7]
```diff
@@ -611,6 +611,85 @@ xy = foo(x, y=3)
...
+
+    bar = 2
+
+    baz = 3
+    ```
+
+- Use blank-lines to separate different multi-line blocks.
+
+    ```julia
+    # Yes
```
Comment (omus) [inline README.md:633]:
```suggestion
    # Yes:
```

File: README.md [hunk 8]
```diff
@@ -611,6 +611,85 @@ xy = foo(x, y=3)
...
+    if foo
+      println("Hi")
+    end
+
+    for i in 1:10
+      println(i)
+    end
+
+    # No
```
Comment (omus) [inline README.md:642]:
```suggestion
    # No:
```

File: README.md [hunk 9]
```diff
@@ -611,6 +611,85 @@ xy = foo(x, y=3)
...
+    end
+    for i in 1:10
+        println(i)
+    end
+    ```
+
+- After a function definition, and before an end statement do not include a line break
+
+    ```julia
+    # Yes
```
Comment (omus) [inline README.md:654]:
```suggestion
    # Yes:
```

File: README.md [hunk 10]
```diff
@@ -611,6 +611,85 @@ xy = foo(x, y=3)
...
+    function foo(bar::Int64, baz::Int64)
+        return bar + baz
+    end
+
+    # No
```
Comment (omus) [inline README.md:659]:
```suggestion
    # No:
```

File: README.md [hunk 11]
```diff
@@ -611,6 +611,85 @@ xy = foo(x, y=3)
...
+    function foo(bar::Int64, baz::Int64)
+
+        return bar + baz
+    end
+
+    # No
```
Comment (omus) [inline README.md:665]:
```suggestion
    # No:
```

**Pull request merged**